Src/Support/Google.Apis.Auth/OAuth2/GoogleCredential.cs

```diff
@@ -104,23 +126,44 @@ public static Task<GoogleCredential> GetApplicationDefaultAsync()
         /// Console or a stored user credential using the format supported by the Cloud SDK.
         /// </para>
         /// </summary>
-        public static GoogleCredential FromStream(Stream stream)
+        public static GoogleCredential FromStream(Stream stream, IEnumerable<string> scopes) =>
+            Scoped(defaultCredentialProvider.CreateDefaultCredentialFromStream(stream), scopes);
+
+        public static GoogleCredential FromStream(Stream stream) => FromStream(stream, Enumerable.Empty<string>());
+
+        public static GoogleCredential FromStream(Stream stream, string scope1, params string[] scopes) =>
+            FromStream(stream, new[] { scope1.ThrowIfNull(nameof(scope1)) }.Concat(scopes.ThrowIfNull(nameof(scopes))));
+
+        public static GoogleCredential FromFile(string path, IEnumerable<string> scopes)
         {
-            return defaultCredentialProvider.CreateDefaultCredentialFromStream(stream);
+            using (var f = File.OpenRead(path))
```

Comment:
Don't think so. `File.OpenRead` is defined as `return new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read)` (See [File.cs](https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/File.cs#L338)). Which is what we'd have coded by hand.